lib/app/features/feed/create_post/views/pages/create_post_modal/create_post_modal.dart

```diff
@@ -29,64 +32,84 @@ class CreatePostModal extends HookWidget {
     final textEditorController = useQuillController();
     final hasContent = useTextEditorHasContent(textEditorController);
 
-    return SheetContent(
-      topPadding: 0,
-      body: Column(
-        children: [
-          NavigationAppBar.modal(
-            title: Text(context.i18n.create_post_modal_title),
+    return BackHardwareButtonInterceptor(
+      onBackPress: (context) async {
+        await showSimpleBottomSheet<void>(
+          context: context,
+          child: CancelCreationModal(
+            title: context.i18n.cancel_creation_post_title,
+            onCancel: () => Navigator.of(context).pop(),
           ),
-          Expanded(
-            child: SingleChildScrollView(
-              child: ScreenSideOffset.small(
-                child: Column(
-                  children: [
-                    Row(
-                      crossAxisAlignment: CrossAxisAlignment.start,
-                      children: [
-                        Avatar(
-                          size: 30.0.s,
-                          imageUrl:
-                              'https://ice-staging.b-cdn.net/profile/default-profile-picture-16.png',
-                        ),
-                        SizedBox(width: 10.0.s),
-                        Expanded(
-                          child: TextEditor(
-                            textEditorController,
-                            placeholder: context.i18n.create_post_modal_placeholder,
+        );
+      },
+      child: SheetContent(
+        topPadding: 0,
+        body: Column(
+          children: [
+            NavigationAppBar.modal(
+              title: Text(context.i18n.create_post_modal_title),
+              onBackPress: () {
+                showSimpleBottomSheet<void>(
```

Comment:
Could you please extract the handling of `onBackPress` to a method? It has the duplicates here and on 36 line.